compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java

```diff
@@ -1443,13 +1443,15 @@ public BType transform(BLangTupleTypeNode tupleTypeNode, AnalyzerData data) {
             memberTypes.add(type);
         }
 
-        BTypeSymbol tupleTypeSymbol = Symbols.createTypeSymbol(SymTag.TUPLE_TYPE, Flags.asMask(EnumSet.of(Flag.PUBLIC)),
+        EnumSet<Flag> flags = tupleTypeNode.flagSet.contains(Flag.ANONYMOUS) ?
```

Comment:
With this change, the actual flag set will be a subset of the original flags, isn't it? Any reason for this?

Reply:
Only `ANONYMOUS` flag will be added before here right, so no flags will be lost?

Reply:
We anyway hadn't used any other flags as well. I wondered whether propagating all the flags to the type symbol would be feasible?

Reply:
Updated for the tuple transform. Should we change for other such cases here as well?